### PR TITLE
Add admin for dynamics form data lists

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | part of #190
| Related issues/PRs | #190, sulu/sulu#4935
| License | MIT

#### What's in this PR?

This PR add an admin for dynamics form list.

#### Why?

This is a missing feature for 2.0 compatability.

#### Example Usage

```yaml
sulu_form:
    dynamic_lists:
        sulu_page.page_edit_form:
            form:
                property: form
                type: page
```

#### BC Breaks/Deprecations

Parent configuration is new in the config tree.

#### To Do

- [x] Update CHANGELOG.md
- [x] Update documentation
- [x] Add line to UPGRADE.md

### CODE REVIEW

File: Resources/doc/dynamic.md [hunk 1]
```diff
@@ -120,11 +120,11 @@ To visualise a tab in the Sulu template, simply configured the following in your
 ```yml
 sulu_form:
     dynamic_lists:
-        content: # tab navigation key e.g. "content" for sulu pages or "article" for sulu articles.
-            form: # unique key mostly the same as the template key or a combination between template and property key.
-                template: form # template key
-                property: form # form property name
-                type: page # the site type e.g. page, article, … (same as the content type type param)
+        page_form: # unique key mostly the same as the template key or a combination between type, template and property key.
+            parent: sulu_page.page_edit_form
```
Comment (alexander-schranz) [inline Resources/doc/dynamic.md]: parent = old `content` key

File: UPGRADE.md [hunk 2]
```diff
@@ -2,6 +2,35 @@
...
+
+The list tab configuration need the parent route key (can be found in the related admin classes).
+Additionally the layer of the "tab navigation" was removed. 
+
+**Before**
+
+```xml
+sulu_form:
+    dynamic_lists:
+        page:
```
Comment (alexander-schranz) [inline UPGRADE.md]: content:

**Comment by alexander-schranz** [comment]
@wachterjohannes is sulu/sulu#4935 required for this PR?

**Comment by wachterjohannes** [comment]
@alexander-schranz yes it is required - will give you a hint when it is merged

**Comment by alexander-schranz** [comment]
@wachterjohannes min requirement need to be `"sulu/sulu": "^2.1",` then

**Comment by wachterjohannes** [comment]
let us discuss that in the team - we should merge this back to 2.0 i think - otherwise form-bundle wont be released before we release sulu 2.1

**Comment by wachterjohannes** [comment]
@alexander-schranz sulu PR is merged to release/2.0

**Comment by nnatter** [comment]
This leads to a `ParentViewNotFoundException` if a user does not have sufficient permissions to see the parent page. For example, if a user does not have any webspace permissions, there is no `sulu_page.page_edit_form` view present and therefore the config request fails with a `ParentViewNotFoundException`: https://github.com/sulu/sulu/blob/master/src/Sulu/Bundle/PageBundle/Admin/PageAdmin.php#L172

**Comment by wachterjohannes** [comment]
@alexander-schranz ready to review again

@nnatter we will discuss that tomorrow and fix this in a following PR

**Pull request merged**